test/cpp/tensorexpr/test_kernel.cpp

```diff
@@ -523,5 +523,231 @@ void testKernelSumMultipleAxes() {
   }
 }
 
+// This test and the following ones testing Softmax only tests with dim set
+// to one of the valid input dimensions. It does not test with dim=None
+// because that is supposed to be deprecated.
+void testKernelSoftmax2D() {
+  const auto graph_template = R"IR(
+      graph(%0 : Float(5:3, 3:1, device=cpu)):
+        %1 : int = prim::Constant[value=${dim}]()
+        %2 : int = prim::Constant[value=7]()
+        %3 : Tensor = aten::softmax(%0, %1, %2)
+        return (%3))IR";
+
+  auto a = iotaTensor({5, 3}, TensorOptions(kCPU).dtype(at::kFloat));
+
+  const std::string& verification_template =
+      R"IR(
+        # CHECK: for (int i0 = 0; i0 < 5
+        # CHECK: for (int i1 = 0; i1 < 3
```

Comment:
Nit: if lines are expected to be next to each other, it might be better to use `CHECK-NEXT`.

Reply:
Done.